File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    int profit=0;
    int buyprice=prices[0];
    for(int i=1;i<pricesSize;i++){
        if(prices[i]<buyprice){
            buyprice=prices[i];
        }
        else{
            int currentprofit=prices[i]-buyprice;
            profit=(currentprofit>profit)?currentprofit:profit;
        }
    }
    return profit;
    
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxprofit=0;
        int min=prices[0];
        
        for(int i=1;i<prices.length;i++){
            int cur=prices[i]-min;
            maxprofit=Math.max(maxprofit,cur);
            min=Math.min(min,prices[i]);
        }
        return maxprofit;
    }
}